src/main/com/yetanalytics/lrs/pedestal/http/multipart_mixed.cljc

```diff
@@ -18,58 +21,123 @@
+                       (re-pattern
+                        (boundary-pat-mid boundary))
+                       body)
+           [[close-idx close-bound]
+            :as close-re-pos] (u/re-pos
+                               (re-pattern
+                                (boundary-pat-close boundary))
+                               body)
+           all-pos (concat open-re-pos
+                           mid-re-pos
+                           close-re-pos)]
+       (assert-valid (= 1 (count open-re-pos))
+                     "Only one opening boundary can be present"
+                     ::invalid-one-opening-boundary)
+       (assert-valid (<= 1 (count mid-re-pos))
+                     "At least one mid boundary must be present"
+                     ::invalid-at-least-one-mid-boundary)
+       (assert-valid (= 1 (count close-re-pos))
+                     "Only one closing boundary can be present"
+                     ::invalid-one-closing-boundary)
+       (for [[[idx-a
+               bound-a :as a]
+              [idx-b :as b]] (partition 2 1 all-pos)]
+         (subs body
+               (+ idx-a
+                  (count bound-a))
+               idx-b)))))
+
+(defn parse-part [^String part boundary]
+  (assert-valid
+   (not (cs/includes? part boundary))
+   "Multipart parts must not include boundary."
+   ::invalid-no-boundary-in-multipart)
+  (let [[headers-str
+         body-str] (cs/split part #"\r\n\r\n")
+        headers    (parse-body-headers headers-str)
+        #?@(:clj [body-bytes (.getBytes body-str "UTF-8")])]
+
+    {:content-type   (get headers "Content-Type")
+     :content-length #?(:clj (count body-bytes)
+                        :cljs (.-length body-str))
+     :headers        headers
+     :input-stream   #?(:clj (ByteArrayInputStream. body-bytes)
+                        :cljs body-str)}))
 
 (defn parse-parts [#?(:clj ^InputStream in
                       :cljs ^String in)
                    ^String boundary]
   (try
     #?(:clj
-       (let [boundary-pattern (make-boundary-pattern boundary)]
-         (with-open [scanner (Scanner. in)]
-           (assert (.hasNext scanner boundary-pattern)
-                   "No initial multipart boundary.")
-           (into []
-                 (for [file-chunk (iterator-seq (.useDelimiter
-                                                 scanner
-                                                 boundary-pattern))
-                       :let [[headers-str
-                              body-str] (cs/split file-chunk #"\R{2}")
-                             headers    (parse-body-headers headers-str)
-                             body-bytes (.getBytes ^String body-str "UTF-8")]]
-                   {:content-type   (get headers "Content-Type")
-                    :content-length (count body-bytes)
-                    :headers        headers
-                    :input-stream   (ByteArrayInputStream. body-bytes)}))))
+       (with-open [^Scanner scanner (.useDelimiter
```

Comment:
Is there a reason why the clj and cljs implementations of `parse-parts` are so radically different? (Part of it is due to historical reasons in lrs, sure, but I'm sure there are good reasons to do so to this day.)

Reply:
Yeah, the main difference is that the clj version operates on an input stream, so it leverages `java.util.Scanner` to tokenize and scan it w/o loading the whole shebang in memory. On cljs it currently just strings the whole body, so it's just pure regex. In the future I'd like to move the cljs version to use node buffers or the like to do it without a stringed body.

Reply:
Makes sense that we want both implementations to be lazy, but switching cljs over to a lazy impl would be outside the scope of this PR.